utils/mochaRunner/src/main.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  TestExpectation,
+  MochaResults,
+  zTestSuiteFile,
+  zPlatform,
+  TestSuite,
+} from './types.js';
+
+import path from 'path';
+import fs from 'fs';
+import os from 'os';
+import {spawn} from 'node:child_process';
+import {
+  extendProcessEnv,
+  filterByPlatform,
+  prettyPrintJSON,
+  readJSON,
+  filterByParameters,
+  getExpectationUpdates,
+  getSkippedTests,
+} from './utils.js';
+
+async function main() {
+  const testSuiteArgIdx = process.argv.indexOf('--test-suite');
+
+  const platform = zPlatform.parse(os.platform());
+
+  const expectations = readJSON(
+    path.join(process.cwd(), 'test', 'TestExpectations.json')
+  ) as TestExpectation[];
+
+  const parsedSuitesFile = zTestSuiteFile.parse(
+    readJSON(path.join(process.cwd(), 'test', 'TestSuites.json'))
+  );
+
+  let applicableSuites: TestSuite[] = [];
+
+  if (testSuiteArgIdx === -1) {
+    applicableSuites = filterByPlatform(parsedSuitesFile.testSuites, platform);
+  } else {
+    const testSuiteId = process.argv[testSuiteArgIdx + 1];
+    const testSuite = parsedSuitesFile.testSuites.find(suite => {
+      return suite.id === testSuiteId;
+    });
+
+    if (!testSuite) {
+      console.error(`Test suite ${testSuiteId} is not defined`);
+      process.exit(1);
+    }
+
+    if (!testSuite.platforms.includes(platform)) {
+      console.warn(
+        `Test suite ${testSuiteId} is not enabled for your platform. Running it anyway.`
+      );
+    }
+
+    applicableSuites = [testSuite];
+  }
+
+  console.log('Planning to run the following test suites', applicableSuites);
+
+  let fail = false;
+  const recommendations = [];
+  try {
+    for (const suite of applicableSuites) {
+      const parameters = suite.parameters;
+
+      const applicableExpectations = filterByParameters(
+        filterByPlatform(expectations, platform),
+        parameters
```

Comment:
```suggestion
        parameters,
```

Reply:
I think our style guide does not require , and automatically removes it